models/coded.go

```diff
@@ -38,6 +38,14 @@ func (c *Coded) AddCodeIfPresent(codeElement xml.Node) {
 		codeSystem = CodeSystemFor(codeElement.Attribute("codeSystem").String())
 	}
 
+	//extract nullFlavor from attribute if it is NA
+	nullFlavorAttribute := codeElement.Attribute("nullFlavor")
+	nullFlavorValueSet := codeElement.Attributes()
+	if nullFlavorAttribute != nil && nullFlavorValueSet["valueSet"] != nil {
+		code = nullFlavorValueSet["valueSet"].String()
+		codeSystem = nullFlavorAttribute.String()
```

Comment:
The naming convention in HDS for this "codeSystem" was "NA_VALUESET".  Currently this code would just assign "NA" as the codeSystem.